crates/floresta-chain/src/pruned_utreexo/chain_state.rs

```diff
@@ -786,25 +772,23 @@ impl<PersistedState: ChainStore> BlockchainInterface for ChainState<PersistedSta
         &self,
         block: &Block,
         proof: Proof,
-        inputs: HashMap<OutPoint, TxOut>,
+        inputs: UtxoMap,
         del_hashes: Vec<sha256::Hash>,
-        acc: Stump,
     ) -> Result<(), Self::Error> {
         // verify the proof
         let del_hashes = del_hashes
             .iter()
             .map(|hash| NodeHash::from(hash.as_byte_array()))
             .collect::<Vec<_>>();
-
-        if !acc.verify(&proof, &del_hashes)? {
+        let acc = self.acc();
+        if acc.verify(&proof, &del_hashes)? {
```

Comment:
Actually, this check isn't needed at all, in rustreexo 0.3 modify will return an error if the proof is invalid.

Reply:
But in `BlockchainInterface::validate_block` we just need to verify the acc without updating it. We modify the acc via `connect_block` (which uses the non-trait validate method).

Reply:
I think this can be removed from this function, and only deal with acc on `update_acc`. The reason I've changed this is that both modify and verify calls rustreexo's most computationally heavy function (`calculate_hashes`). We get like a 50% or more speedup by not calling `verify` during IBD.

Reply:
I think there may be a confusion here:

- `connect_block` (used in IBD): calls `validate_block` and then `update_acc`
- `BlockchainInterface::validate_block` (validation tool, not IBD): calls `verify` acc and then `validate_block`
- `validate_block`: common inner method

So we are not dealing with the acc twice. Is this what you were thinking?